roles/ceph-nfs/tasks/pre_requisite.yml

```diff
@@ -1,41 +1,51 @@
 ---
-- name: create nfs gateway directories
-  file:
-    path: "{{ item }}"
-    state: directory
-    owner: "ceph"
-    group: "ceph"
-    mode: "0755"
-  with_items:
-    - /var/lib/nfs/ganesha
-    - /var/run/ganesha
-
 - name: create rgw nfs user
   command: radosgw-admin --cluster {{ cluster }} user create --uid={{ ceph_nfs_rgw_user }} --display-name="RGW NFS User"
   register: rgwuser
+  delegate_to: "{{ groups[mon_group_name][0] }}"
   when: nfs_obj_gw
 
 - name: set access key
   set_fact:
     ceph_nfs_rgw_access_key: "{{ (rgwuser.stdout | from_json)['keys'][0]['access_key'] }}"
-  when: nfs_obj_gw
+  delegate_to: "{{ groups[mon_group_name][0] }}"
```

Comment:
@leseb the reason this is delegated to the monitor is because it uses the rgwuser register in the "create rgw nfs user" above.